src/Groups/MatGrp.jl

```diff
@@ -0,0 +1,790 @@
+import AbstractAlgebra: MatElem, matrix, MatSpace, parent_type, Ring, RingElem
+import Hecke: base_ring, det, fmpz, fq_nmod, FqNmodFiniteField, nrows, tr, trace
+import GAP: FFE
+
+export
+    general_linear_group,
+    mat_elem_type,
+    matrix_group,
+    MatrixGroup,
+    MatrixGroupElem,
+    omega_group,
+    orthogonal_group,
+    ring_elem_type,
+    special_linear_group,
+    special_orthogonal_group,
+    special_unitary_group,
+    symplectic_group,
+    unitary_group,
+    GL, GO, GU, SL, SO, Sp, SU
+
+
+
+abstract type AbstractMatrixGroupElem <: GAPGroupElem{GAPGroup} end
+
+# NOTE: always defined are deg, ring and at least one between { X, gens, descr }
+# NOTE: the field mat_iso are always defined if the field X is
+"""
+    MatrixGroup{RE<:RingElem, T<:MatElem{RE}} <: GAPGroup
+Type of groups `G` of `n x n` matrices over the ring `R`, where `n = degree(G)` and `R = base_ring(G)`.
+
+At the moment, only rings of type `FqNmodFiniteField` are supported.
+"""
+mutable struct MatrixGroup{RE<:RingElem, T<:MatElem{RE}} <: GAPGroup
+   deg::Int
+   ring::Ring
+   X::GapObj
+   gens::Vector{<:AbstractMatrixGroupElem}
+   descr::Symbol                       # e.g. GL, SL, symbols for isometry groups
+   mat_iso::GenMatIso
+   order::fmpz
+   AbstractAlgebra.@declare_other
+
+   MatrixGroup(m::Int, F::Ring) = new{elem_type(typeof(F)), dense_matrix_type(elem_type(typeof(F)))}(m,F)
+
+end
+
+MatrixGroup{RE,T}(m::Int, F::Ring) where {RE,T} = MatrixGroup(m,F)
```

Comment:
This is backwards: the parametric constructor should be "inner" and the default, and the non-parametric one should call it. With your construction, the parameters are ignored!

Reply:
The line
> MatrixGroup{RE,T}(m::Int, F::Ring) where {RE,T} = new{RE,T}(m,F)

defines the default constructor. Should I add a check to guarantee the parametric type is correct? This is something like
> @ assert element_type(F) <: RE